### DISCUSSION TITLE
How can I fit a polynomial with a mask on signal dimension?

### DISCUSSION THREAD

**Discussion by PZH525** [hyperspy/hyperspy#3159] [Q&A] [answered]
Hi there, 

Currently I'm trying to fit my background as a polynomial function, and I tried to use mask function in multifit(). However, after checking the source code, I find that the mask for multifit() can be only apply on the navigation dimension. 

Is there a way to apply mask on the signal dimension? I also tried the remove_signal_range, but this command will render discontinues fit eventually which is not what I want.(see pic). 
<img width="321" alt="Fig" src="https://github.com/hyperspy/hyperspy/assets/69710448/275a6ebc-1081-47f1-9614-35a4e5e55bc1">
How can I get a continues fit with some data point excluded?

**Comment by ericpre** [comment, marked as answer]
`remove_signal_range` will exclude the data from the fit, i.e. mask them and `reset_signal_range` will reset the signal range.
https://hyperspy.org/hyperspy-doc/current/api/hyperspy.models.model1d.html#hyperspy.models.model1d.Model1D.reset_signal_range

After fitting, you need to call `reset_signal_range` to displayed the whole range.

**Reply by PZH525** [reply]
Thank you for your reply!
